app/src/main/kotlin/compiler/semantic_analysis/BuiltinFunctions.kt

```diff
@@ -0,0 +1,75 @@
+package compiler.semantic_analysis
+
+import compiler.ast.Expression
+import compiler.ast.Function
+import compiler.ast.Program
+import compiler.ast.Statement
+import compiler.ast.Type
+import compiler.ast.Variable
+import compiler.common.reference_collections.ReferenceSet
+import compiler.common.reference_collections.referenceHashSetOf
+
+val functionNapisz = Function(
+    "napisz",
+    listOf(Function.Parameter("wartość", Type.Number, null)),
+    Type.Unit,
+    listOf()
+)
+
+val builtinFunctionsByName = mapOf(functionNapisz.name to functionNapisz)
+
+fun getUsedBuiltinFunctions(program: Program): ReferenceSet<Function> {
+    val result = referenceHashSetOf<Function>()
+
+    fun analyzeNode(node: Any) {
```

Comment:
It would be nicer to avoid Any and do it "by type" like in e.g. callGraph construction.

Reply:
Added with exhaustive `when` expressions